api/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -51,6 +51,12 @@ type OpenTelemetryCollectorSpec struct {
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 
+	// VolumeClaimTemplates will provide stable storage using PersistentVolumes.
+	// +optional
+	// +listType=atomic
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	VolumeClaimTemplates []v1.PersistentVolumeClaim `json:"volumeClaimTemplates,omitempty"`
```

Comment:
Should this be PersistentVolumeClaimTemplate?

Reply:
Also interesting that in the description of v1.PersistentVolumeClaimTemplate it says for ObjectMeta:

```golang
	// May contain labels and annotations that will be copied into the PVC
	// when creating it. No other fields are allowed and will be rejected during
	// validation.
	//
	// +optional
	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Reply:
I wonder if that has something to do with why name is not propagated